src/main/java/org/opensearch/knn/index/codec/transfer/VectorTransferByteToFloat.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.knn.index.codec.transfer;
+
+import org.opensearch.knn.index.codec.util.SerializationMode;
+import org.opensearch.knn.jni.JNICommons;
+
+import java.io.ByteArrayInputStream;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Vector transfer for byte vectors by casting them as float vectors
+ */
+public class VectorTransferByteToFloat extends VectorTransfer {
```

Comment:
Why we need to cast then for vector transfer?

Reply:
As we are using scalar quantizer behind the scenes which expects input as float vector. So, while deserializing it back to byte vector casting it as float vector and then transferring it